web/siteplan/src/util/Controls/ExportControl.ts

```diff
@@ -50,182 +70,456 @@ export default class ExportControl extends Control {
+      const currentSourceMap = store.state.selectedSourceMap
+      store.commit('setSourceMap', new EmptyMap().getIdentifier())
+      const visibleLayers = store.state.featureLayers.filter(layer => layer.getVisible())
+      visibleLayers.forEach(layer => layer.setVisible(false))
+      setMapScale(this.map.getView(), 1000)
+      this.map.getView().setRotation(0)
+      const resolution = this.map.getView().getResolution()
+
+      const exportCanvases = await this.getSiteplanSheetcutExportCanvas(
+        sheetCutFeatures,
+        visibleLayers,
+        resolution ?? 1
+      )
+
+      store.commit('setSourceMap', currentSourceMap)
+      visibleLayers.forEach(layer => layer.setVisible(true))
+      this.map.getView().setRotation(originalRotation)
+      this.map.getView().setZoom(originalZoomLvl ?? 10)
+      this.map.getView().setCenter(originalViewCenter)
+
+      result.push(...exportCanvases.filter(canvas => canvas !== null))
+    } else {
+      const currentCanvas = await this.getCurrentCanvas()
+      result.push(currentCanvas)
+    }
+
+    const link = document.createElement('a')
+    result.filter(c => c !== undefined).forEach((c, index) => {
+      link.setAttribute('download', 'siteplan' + (index == 0 ? '' : index))
+      if (link) {
+        link.href = c.toDataURL()
+        link.click()
+      }
+    })
+    this.lockMapDuringExport(false)
+  }
+
+  private async getCurrentCanvas () {
+    const viewportSize = this.map.getSize()
+    if (!viewportSize) {
+      return
+    }
+
+    const canvas = document.createElement('canvas')
+    canvas.width = viewportSize[0]
+    canvas.height = viewportSize[1]
+
+    const context = canvas.getContext('2d')
+    if (context == null) {
       return
     }
 
-    const visbleLayer = store.state.featureLayers.filter(layer => layer.getVisible() === true)
-    visbleLayer.forEach(layer => layer.setVisible(false))
-    const originalRotation = this.map.getView().getRotation()
-    const originalZoomLvl = this.map.getView().getZoom()
-    const originalViewCenter = this.map.getView().getCenter()
+    context.fillStyle = 'white'
+    context.fillRect(0, 0, canvas.width, canvas.height)
+    const canvasLayers = document.querySelectorAll('.ol-layer canvas')
+    canvasLayers.forEach(layer => {
+      const mapCanvas = layer as HTMLCanvasElement
+      if (mapCanvas.width < 0) {
+        return
+      }
+
+      const matrix = this.getMapTransformMatrix(mapCanvas)
+      context.transform(matrix[0], matrix[1], matrix[2], matrix[3], matrix[4], matrix[5])
 
+      context.drawImage(mapCanvas, 0, 0)
+      context.setTransform(1, 0, 0, 1, 0 ,0)
+    })
+    await new Promise(res => setTimeout(res, 2000))
+    return canvas
+  }
+
+  private async getSiteplanSheetcutExportCanvas (
+    sheetCutFeatures: Feature<Geometry>[],
+    visibleLayers: NamedFeatureLayer[],
+    resolution: number
+  ) {
+    const result: HTMLCanvasElement[] = []
     for (const feature of sheetCutFeatures) {
```

Comment:
@TruongQuangSB To not conflict with all the other features I would sugest to rename it to

```suggestion
    for (const sheetCutFeature of sheetCutFeatures) {
```